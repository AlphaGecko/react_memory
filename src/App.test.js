import React from 'react'

import { shallow } from 'enzyme';
import { expect } from 'chai'; // Utilisation de la syntaxe chai
import sinon from 'sinon'

import GuessCount from './GuessCount';
import Card from './Card'
import App from './App'

describe('<Card/>', () => {
  it('Appel l\'événement "onClick" quand une carte est cliquée', () => {
      const onClick = sinon.spy()
      const wrapper = shallow(
      <Card card="😁" feedback="hidden" index={0} onClick={onClick} />
      )

      wrapper.simulate('click')
      expect(onClick).to.have.been.calledWith(0)
  })

  it('Comparaison snapshot', () => {
    const onClick = sinon.spy()
    const wrapper = shallow(
    <Card card="😁" feedback="hidden" index={0} onClick={onClick} />
    )

    expect(wrapper).to.matchSnapshot()
  })
})

describe('<App />', () => {
  it('Ne crash pas !', () => {
    const wrapper = shallow(<App />)
  
    expect(wrapper).to.contain(<GuessCount guesses={0} />)
  })

  it('Possède 36 cartes', () => {
    const wrapper = shallow(<App />)
  
    expect(wrapper.find('Card')).to.have.length(36)
  })
})