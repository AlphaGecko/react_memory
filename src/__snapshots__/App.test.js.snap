// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> Comparaison snapshot 1`] = `
<div
  className="card hidden"
  onClick={[Function]}
>
  <span
    className="symbol"
  >
    ‚ùì
  </span>
</div>
`;
